lib/galaxy/managers/history_contents.py

```diff
@@ -531,26 +531,11 @@ class HistoryContentsFilters(
 
     def parse_query_filters_with_relations(self, query_filters: ValueFilterQueryParams, history_id):
         """Parse query filters but consider case where related filter is included."""
-        if query_filters.q and query_filters.qv and "related-eq" in query_filters.q:
-            qv_index = query_filters.q.index("related-eq")
-            qv_hid = query_filters.qv[qv_index]
```

Comment:
As a general rule of thumb (not a hard one) I'd avoid the refactoring within bugfixes. You could just cast to int here and if that fails raise an exception.

Reply:
(This should go to 23.0)

Reply:
Ok, can I make a separate PR for the refactoring of the `get_query_filters_with_relations` function, that'll just target dev?

Reply:
yes, thank you!